spec/models/spree/order_spec.rb

```diff
@@ -169,10 +169,16 @@
       expect(order.shipment_state).to eq 'ready'
     end
 
-    it "should send an order confirmation email" do
-      expect do
-        order.finalize!
-      end.to enqueue_job ConfirmOrderJob
+    it "sends confirmation emails to both the user and the shop owner" do
+      customer_confirm = double(:confirm_email_for_customer)
+      shop_confirm = double(:confirm_email_for_shop)
+
+      expect(Spree::OrderMailer).to receive(:confirm_email_for_customer).and_return(customer_confirm)
+      expect(Spree::OrderMailer).to receive(:confirm_email_for_shop).and_return(shop_confirm)
+      expect(customer_confirm).to receive(:deliver_later)
+      expect(shop_confirm).to receive(:deliver_later)
```

Comment:
This test code is basically a duplication of the real code. You're just testing that you typed everything correctly but we are not testing that a job is added to the queue or that the job actually sends an email.

This spec passes even though our code fails. I think that we should test with real objects if possible.